geopandas/tests/test_cmapplotting.py

```diff
@@ -0,0 +1,112 @@
+        values = np.arange(self.N)
+        levels = ['1','1','2','4','5','1','5','4','3','2']
+        self.df = GeoDataFrame({'geometry': self.points, 'values': values, 'levels': levels})
+
+    def test_customnorm(self):
+        # test color without additional parameters
+        ax = self.df.plot(column='values')
+        actual_color = ax.collections[0].get_facecolors()
+        cmap = ax.collections[0].get_cmap()
+        expected_color = cmap(np.arange(self.N) / (self.N - 1))
+        np.testing.assert_array_equal(actual_color, expected_color)
+
+        # test color with norm
+        cmap = mpl.cm.Greys
+        norm = mpl.colors.BoundaryNorm([1,3,6,8], cmap.N)
+        ax = self.df.plot(column='values', norm = norm)
+        actual_color = ax.collections[0].get_facecolors()
+        cmap = ax.collections[0].get_cmap()
+        expected_color = cmap(norm(list(self.df['values'])))
+        np.testing.assert_array_equal(actual_color, expected_color)
+
+    def test_minmax(self):
+        # test colorbar min,max without additional parameters
+        ax = self.df.plot(column='values', legend=True)
+        colorbar_color = ax.get_figure().axes[1].collections[0].get_clim()
+        point_color = ax.collections[0].get_clim()
+        assert colorbar_color == point_color
+
+        # test colorbar min,max with norm
+        cmap = mpl.cm.Greys
+        norm = mpl.colors.BoundaryNorm([1,3,6,8], cmap.N)
+        ax = self.df.plot(column='values', legend=True, norm = norm)
+        colorbar_color = ax.get_figure().axes[1].collections[0].get_clim()
+        point_color = ax.collections[0].get_clim()
+        assert colorbar_color == point_color
+
+    def test_subsets(self):
+        # test min, max of subplot and plot without additional parameters
+        ax = self.df.plot(column='values')
+        actual_lim = ax.collections[0].get_clim()
+        actual_color = ax.collections[0].get_facecolors()
+
+        lvl1 = self.df['levels'] == '5'
+        ax1 = self.df[lvl1].plot(column='values')
+        actual_lim1 = ax1.collections[0].get_clim()
+        actual_color1 = ax1.collections[0].get_facecolors()
+
+        # Limits must not be equal
+        assert actual_lim != actual_lim1
+
+        # Colors of min and max for both the plot are equal(which is not correct)
+        np.testing.assert_array_equal(actual_color[0],actual_color1[0])
+        np.testing.assert_array_equal(actual_color[-1],actual_color1[-1])
+
+        # test min,max for subplots and plot with norm
+        # obtain min, max and actual colors from axis
+        cmap = mpl.cm.Greys
+        norm = mpl.colors.BoundaryNorm([1,3,6,7,9], cmap.N)
+        ax = self.df.plot(column='values',  legend=True, norm = norm)
+        actual_lim = ax.collections[0].get_clim()
+        actual_color = ax.collections[0].get_facecolors()
+        cmap = ax.collections[0].get_cmap()
+        expected_color = cmap(norm(list(self.df['values'])))
```

Comment:
What is the different of this part with what you are testing in `test_customnorm` above?

Reply:
Can you answer to this comment?

Reply:
First, I'm trying to get the limits and actual color from the original dataframe, then I'm doing the same for the subset, so that I can check the respective limits and the color of the maximum and minimum value. I think ```expected_color``` is not required for this test.